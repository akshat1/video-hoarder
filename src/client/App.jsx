import LoginForm from "./components/LoginForm.jsx";
import Main from "./components/Main.jsx";
import { getHistory } from "./history";
import { initializeClient } from "./redux/actions";
import { isFetchingUser, isLoggedIn, isUserFetchDone } from "./selectors";
import { getTheme } from "./theme";
import { getURL } from "./util.js";
import { useMediaQuery } from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import { makeStyles, ThemeProvider } from "@material-ui/styles";
import { ConnectedRouter } from "connected-react-router";
import PropTypes from "prop-types";
import React, { useEffect } from "react";
import { hot } from "react-hot-loader";
import { connect } from "react-redux";
import { Route,Switch } from "react-router";

const useStyles = makeStyles(() => ({
  container: {},
  loginForm: {
    position: "absolute",
    top: "20%",
    left: "50%",
    transform: "translate(-50%, -50%)",
  },
}));

const App = (props) => {
  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
  const classes = useStyles();
  const {
    userFetchDone,
    initializeClient,
    loggedIn,
  } = props;

  useEffect(
    () => { initializeClient(); },
    [userFetchDone],
  );

  return (
    <ConnectedRouter history={getHistory()}>
      <ThemeProvider theme={getTheme(prefersDarkMode)}>
        <CssBaseline />
        <Switch>
          <Route exact path={getURL("/login")}>
            <LoginForm className={classes.loginForm}/>
          </Route>
          <Route path={getURL("*")}>
            <Choose>
              <When condition={loggedIn}>
                <Main />
              </When>
            </Choose>
          </Route>
        </Switch>
      </ThemeProvider>
    </ConnectedRouter>
  );
};

App.propTypes = {
  fetchingUser: PropTypes.bool,
  initializeClient: PropTypes.func,
  loggedIn: PropTypes.bool,
  userFetchDone: PropTypes.bool,
};

/**
 * @private
 * @returns {boolean}
 */
const isDevMode = () => process.env.NODE_ENV === "development";

const stateToProps = state => ({
  fetchingUser: isFetchingUser(state),
  loggedIn: isLoggedIn(state),
  userFetchDone: isUserFetchDone(state),
});

const dispatchToProps = { initializeClient };

export { App };
const ConnectedApp = connect(stateToProps, dispatchToProps)(App);
// eslint-disable-next-line import/no-default-export
export default isDevMode() ? hot(module)(ConnectedApp) : ConnectedApp;
