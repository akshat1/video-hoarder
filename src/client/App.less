@import (reference) "./style/vars.less";

html, body {
  height: 100vh;
  width: 100vw;
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-size: ~"@{mobile-font-size}px";
  font-family: "Open Sans", "sans serif";

  @media @desktop {
    font-size: ~"@{desktop-font-size}px";
  }
}
