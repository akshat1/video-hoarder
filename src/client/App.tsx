import { Main } from "./Main";
import React, { Fragment } from "react";

export const App:React.FunctionComponent = () => {
  return (
    <Fragment>
      <Main />
    </Fragment>
  );
};
