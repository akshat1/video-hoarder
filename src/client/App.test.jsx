/** @jest-environment jsdom */
import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  test('App matches snapshot', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
