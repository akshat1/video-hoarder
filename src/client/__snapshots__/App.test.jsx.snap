// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App matches snapshot 1`] = `
<Connect(ConnectedRouterWithContext)
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <div>
    <Switch>
      <Route
        path="/login"
      >
        <Connect(LoginForm) />
      </Route>
      <Route
        path="/"
      />
    </Switch>
  </div>
</Connect(ConnectedRouterWithContext)>
`;
